Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,332 @@
+                            tf.GetTranslation().GetY(),
+                            tf.GetTranslation().GetZ(),
+                            tf.GetRotation().GetX(),
+                            tf.GetRotation().GetY(),
+                            tf.GetRotation().GetZ(),
+                            tf.GetRotation().GetW());
+                        transformInterface->SetWorldTM(tf);
+                    }
+                    else
+                    {
+                        AZ_TracePrintf(
+                            "CreatePrefabFromURDF", "Setting transform failed: %s does not have transform interface", name.c_str());
+                    }
+                }
+            }
+        }
+
+        // set hierarchy
+        for (const auto& [name, link_ptr] : links)
+        {
+            const auto this_entry = created_links.at(name);
+            if (!this_entry.IsSuccess())
+            {
+                AZ_TracePrintf("CreatePrefabFromURDF", "Link %s creation failed", name.c_str());
+                continue;
+            }
+            auto parent_ptr = link_ptr->getParent();
+            if (!parent_ptr)
+            {
+                AZ_TracePrintf("CreatePrefabFromURDF", "Link %s has no parents", name.c_str());
+                continue;
+            }
+            AZStd::string parent_name(parent_ptr->name.c_str(), parent_ptr->name.size());
+            const auto parent_entry = created_links.find(parent_name);
+            if (parent_entry == created_links.end())
+            {
+                AZ_TracePrintf("CreatePrefabFromURDF", "Link %s has invalid parent name %s", name.c_str(), parent_name.c_str());
+                continue;
+            }
+            if (!parent_entry->second.IsSuccess())
+            {
+                AZ_TracePrintf(
+                    "CreatePrefabFromURDF", "Link %s has parent %s which has failed to create", name.c_str(), parent_name.c_str());
+                continue;
+            }
+            AZ_TracePrintf(
+                "CreatePrefabFromURDF",
+                "Link %s setting parent to %s",
+                this_entry.GetValue().ToString().c_str(),
+                parent_entry->second.GetValue().ToString().c_str());
+            AZ_TracePrintf("CreatePrefabFromURDF", "Link %s setting parent to %s", name.c_str(), parent_name.c_str());
+            auto* entity = AzToolsFramework::GetEntityById(this_entry.GetValue());
+            entity->Activate();
+            AZ::TransformBus::Event(this_entry.GetValue(), &AZ::TransformBus::Events::SetParent, parent_entry->second.GetValue());
+            entity->Deactivate();
+        }
+
+        // create joint
+        auto joints = Utils::GetAllJoints(m_model->root_link_->child_links);
+        for (const auto& [name, joint_ptr] : joints)
+        {
+            AZ_Assert(joint_ptr, "joint %s is null", name.c_str());
+            AZ_TracePrintf(
+                "CreatePrefabFromURDF",
+                "Creating joint %s : %s -> %s",
```

Comment:
This trace printf format string needs a newline at the end.
```suggestion
                "Creating joint %s : %s -> %s\n",
```